janusgraph-core/src/main/java/org/janusgraph/graphdb/types/StandardPropertyKeyMaker.java

```diff
@@ -34,12 +35,14 @@
 public class StandardPropertyKeyMaker extends StandardRelationTypeMaker implements PropertyKeyMaker {
 
     private Class<?> dataType;
+    private boolean cardinalityIsSet;
 
     public StandardPropertyKeyMaker(StandardJanusGraphTx tx, String name, IndexSerializer indexSerializer,
                                     final AttributeHandler attributeHandler) {
         super(tx, name, indexSerializer, attributeHandler);
         dataType = null;
         cardinality(Cardinality.SINGLE);
+        cardinalityIsSet = false;
```

Comment:
Also, not your code but `dataType = null` is also a redundant initialization. By default it is `null`.
Variables must be assigned with values only when they are created in methods. Otherwise they have default values.

Reply:
> (nitpick) redundant initialization. By default `cardinalityIsSet` is already `false`.

cardinality overwrites cardinalityIsSet to false. So, I have to reset it.

Reply:
Ah, yes I didn't see it